functions/geometry_kube.zsh

```diff
@@ -3,15 +3,28 @@
 (( $+commands[kubectl] )) || return
 
 geometry_kube() {
-  ( ${GEOMETRY_KUBE_PIN:=false} ) || [[ -f ~/.kube/config ]] || return
+  ( ${GEOMETRY_KUBE_PIN:=false} ) || [[ ! -z "${KUBECONFIG}" ]] || return
 
-  GEOMETRY_KUBE=$(ansi ${GEOMETRY_KUBE_COLOR:=blue} ${GEOMETRY_KUBE_SYMBOL:="⎈"})
+  local geometry_kube=$(ansi ${GEOMETRY_KUBE_COLOR:=blue} ${GEOMETRY_KUBE_SYMBOL:="⎈"})
+  local geometry_kube_separator=${GEOMETRY_KUBE_SEPARATOR:="|"}
 
-  [[ $(kubectl version --client --short) =~ 'Client Version: ([0-9a-zA-Z.]+)' ]]
-  local version=$match[1]
+  if ( ${GEOMETRY_KUBE_VERSION:=true} ); then
+    [[ $(kubectl version --client --short) =~ 'Client Version: ([0-9a-zA-Z.]+)' ]]
+    geometry_kube+=($match[1])
+  fi
 
-  local context="$(kubectl config current-context 2> /dev/null)"
-  local namespace="$(kubectl config view -o "jsonpath={.contexts[?(@.name==\"$context\")].context.namespace}" 2> /dev/null)"
+  local geometry_context="$(kubectl config current-context 2> /dev/null)"
+  local geometry_namespace="$(kubectl config view -o "jsonpath={.contexts[?(@.name==\"$context\")].context.namespace}" 2> /dev/null)"
```

Comment:
maybe `kube_context` and `kube_namespace`?

Reply:
Though I can see an argument for this being geometry specific, not kube_specific. Either way, these variables should be 'hidden' unless something breaks, in which case letting the person know 'this is not kube broken, its your prompt' is less panic-inducing.